battlepass/src/benchmarking.rs

```diff
@@ -0,0 +1,171 @@
+	<T as pallet::Config>::Currency::deposit(T::NativeTokenId::get(), &account_id, balance_amount)?;
+	<T as pallet::Config>::Currency::deposit(T::ProtocolTokenId::get(), &account_id, balance_amount)?;
+	Ok(())
+}
+
+/// Fund account with tokens, needed for org and battlepass interactions
+fn get_funded_caller<T: Config>() -> Result<T::AccountId, DispatchError> {
+    let caller: T::AccountId = whitelisted_caller();
+    fund_account::<T>(&caller)?;
+
+    Ok(caller)
+}
+
+
+fn get_org<T: Config>(caller: T::AccountId) -> T::Hash {
+    let org_id = T::ControlBenchmarkHelper::create_org(caller.clone()).unwrap();
+    let treasury_id = T::Control::org_treasury_account(&org_id).unwrap();
+	fund_account::<T>(&treasury_id).unwrap();
+
+    org_id
+}
+
+fn get_battlepass<T: Config>(caller: T::AccountId, org_id: T::Hash) -> T::Hash {
+    let str = BoundedVec::truncate_from(vec![1,2]);
+    let battlepass = types::Battlepass {
+        creator: caller.clone(),
+        org_id,
+        name: str.clone(),
+        cid: str.clone(),
+        season: 1,
+        price: 10,
+        collection_id: 0
+    };
+
+    let _ = BPass::<T>::create_battlepass(RawOrigin::Signed(caller.clone()).into(), org_id, battlepass.cid.clone(), battlepass.name.clone(), battlepass.price.clone());
+
+    <T as frame_system::Config>::Hashing::hash_of(&battlepass)
+}
+
+fn activate_bpass<T: Config>(caller: T::AccountId, battlepass_id: T::Hash) {
+    let _ = BPass::<T>::activate_battlepass(RawOrigin::Signed(caller).into(), battlepass_id);
+}
+
+fn claim_bpass<T: Config>(caller: T::AccountId, battlepass_id: T::Hash) {
+    let _ = BPass::<T>::claim_battlepass(RawOrigin::Signed(caller.clone()).into(), battlepass_id, caller);
+}
+
+fn set_bpass_points<T: Config>(caller: T::AccountId, battlepass_id: T::Hash) {
+    let _ = BPass::<T>::set_points(RawOrigin::Signed(caller.clone()).into(), battlepass_id, caller, 10);
+}
+
+fn set_bpass_level<T: Config>(caller: T::AccountId, battlepass_id: T::Hash) {
+    let _ = BPass::<T>::add_level(RawOrigin::Signed(caller.clone()).into(), battlepass_id, 1, 10);
+}
+
+fn get_reward<T: Config>(caller: T::AccountId, battlepass_id: T::Hash) -> T::Hash {
+    let str = BoundedVec::truncate_from(vec![1,2]);
+    let reward = Reward{
+        battlepass_id,
+        name: str.clone(),
+        cid: str.clone(),
+        level: 1,
+        transferable: true,
+        collection_id: 1
+    };
+
+    let _ = BPass::<T>::create_reward(RawOrigin::Signed(caller).into(), battlepass_id, reward.name.clone(), reward.cid.clone(), Some(10), reward.level.clone(), reward.transferable.clone());
+
+    <T as frame_system::Config>::Hashing::hash_of(&reward)
+}
+
+
+benchmarks! {
+
+	create_battlepass {
+        let caller: T::AccountId = get_funded_caller::<T>()?;
+        let org_id = get_org::<T>(caller.clone());
+        let str = BoundedVec::truncate_from(vec![1,2]);
+    }: _(RawOrigin::Signed(caller), org_id, str.clone(), str.clone(), 10)
```

Comment:
Add `verify` statement to check for expected result. Here and for other benchmarks.

Reply:
Will add it in a separate PR.